src/compiler/types.ts

```diff
@@ -4101,19 +4102,28 @@ export type FlowNode =
 
 export interface FlowNodeBase {
     flags: FlowFlags;
-    id?: number; // Node id used by flow type cache in checker
+    id: number; // Node id used by flow type cache in checker
+    node: unknown; // Node or other data
+    antecedent: FlowNode | FlowNode[] | undefined;

```

Comment:
I'm kind of surprised that this turned out okay; I would have thought these two types would have different shapes, but I guess they're both pointers...

Reply:
They're both heap objects so should get the same hidden class.